src/app/profiles/multipleChoiceEditor.tsx

```diff
@@ -0,0 +1,197 @@
+'use client'
+
+import {Button} from "@/components/ui/button"
+import {
+    Card,
+    CardContent,
+    CardDescription,
+    CardFooter,
+    CardHeader,
+    CardTitle,
+} from "@/components/ui/card"
+import { Label } from "@/components/ui/label"
+import { RadioGroup, RadioGroupItem } from "@/components/ui/radio-group"
+import { Dispatch, SetStateAction, useRef, useState } from "react"
+import { SortableContext, verticalListSortingStrategy, useSortable, arrayMove } from "@dnd-kit/sortable"
+import { DndContext, DragEndEvent, closestCenter } from '@dnd-kit/core'
+import { CSS } from '@dnd-kit/utilities'
+import { Formik, Form, Field, FieldArray, FieldArrayRenderProps, FormikErrors } from 'formik'
+
+type Answer = {
+    id: number
+    label: string
+}
+
+const SortableInput = ({answer, index, answers, setAnswers, arrayHelpers, setFieldValue}: {answer: Answer, index: number, answers: Answer[], setAnswers: Dispatch<SetStateAction<Answer[]>>, arrayHelpers: FieldArrayRenderProps, setFieldValue: (field: string, value: any, shouldValidate?: boolean | undefined) => Promise<void | FormikErrors<{question: string; answers: string[];}>>}) => {
+    const {
+        attributes,
+        listeners,
+        setNodeRef,
+        transform,
+        transition,
+    } = useSortable({id: answer.id})
+
+    const style = {
+        transition,
+        transform: CSS.Transform.toString(transform),
+    }
+
+    return (
+        <div key={index} style={style} {...attributes}>
+            <br></br>
+            <Label>Answer {index + 1}</Label>
+            <div className="grid grid-cols-10 gap-3">
+                { answers.length > 1 ?
+                    <>
+                        <div className="col-span-8">
+                            <Field name={`answers.${index}`} type="text" placeholder="" value={answer.label} className="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50" onChange={(e: any) => {
+                                answer.label = e.target.value
+                                setAnswers(answers.map(el => {
+                                    return el.id === answer.id ? answer : el
+                                }))
+                                setFieldValue(`answers.${index}`, e.target.value)
```

Comment:
I think there's a cooler way to do this by using `useField()` and what it returns instead of `setFieldValue` directly